util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -233,4 +246,68 @@ static URI parseProxyAddress(@Nullable String proxyAddress) {
           String.format("Invalid proxy address '%s'.", proxyAddress), e);
     }
   }
+
+  /** Wrapper class to have socketKeepAlive property while creating the socket */
+  @VisibleForTesting
+  static class SslKeepAliveSocketFactory extends SSLSocketFactory {
+
+    private final SSLSocketFactory wrappedSockedFactory;
+
+    public SslKeepAliveSocketFactory(SSLSocketFactory wrappedSocketFactory) {
+      this.wrappedSockedFactory = wrappedSocketFactory;
+    }
+
+    @Override
+    public String[] getDefaultCipherSuites() {
+      return wrappedSockedFactory.getDefaultCipherSuites();
+    }
+
+    @Override
+    public String[] getSupportedCipherSuites() {
+      return wrappedSockedFactory.getSupportedCipherSuites();
+    }
+
+    @Override
+    public Socket createSocket() throws IOException {
+      return setSocketKeepAlive(wrappedSockedFactory.createSocket());
+    }
+
+    @Override
+    public Socket createSocket(Socket s, InputStream consumed, boolean autoClose)
+        throws IOException {
+      return setSocketKeepAlive(wrappedSockedFactory.createSocket(s, consumed, autoClose));
+    }
+
+    @Override
+    public Socket createSocket(Socket s, String host, int port, boolean autoClose)
+        throws IOException {
+      return setSocketKeepAlive(wrappedSockedFactory.createSocket(s, host, port, autoClose));
+    }
+
+    public Socket createSocket(String host, int port) throws IOException {
+      return setSocketKeepAlive(wrappedSockedFactory.createSocket(host, port));
+    }
+
+    public Socket createSocket(InetAddress address, int port) throws IOException {
+      return setSocketKeepAlive(wrappedSockedFactory.createSocket(address, port));
+    }
+
+    public Socket createSocket(String host, int port, InetAddress clientAddress, int clientPort)
+        throws IOException {
+      return setSocketKeepAlive(
+          wrappedSockedFactory.createSocket(host, port, clientAddress, clientPort));
+    }
+
+    public Socket createSocket(
+        InetAddress address, int port, InetAddress clientAddress, int clientPort)
+        throws IOException {
+      return setSocketKeepAlive(
+          wrappedSockedFactory.createSocket(address, port, clientAddress, clientPort));
+    }
+
+    private static Socket setSocketKeepAlive(Socket socket) throws SocketException {
+      socket.setKeepAlive(true);
```

Comment:
Please, add a release note for this in CHANGES.md

Reply:
Done